### PR TITLE
Fix #bytes encoding and add #bytes(WordStack) encoding

### PR DESCRIPTION
**Pull request by tomtomjhj**

### CODE REVIEW

**Comment by daejunpark** [comment]
@tomtomjhj could you explain a bit more about the encoding bug? it's opposite to my intuition. you mean the correct encoding pads zeros in the right? so, `#bytes( 10, 255 )` means that ` 11...11000....000` instead of `000...00011...11`?

**Comment by tomtomjhj** [comment]
@daejunpark [yes](https://solidity.readthedocs.io/en/develop/abi-spec.html#use-of-dynamic-types)

**Comment by daejunpark** [comment]
@yzhang90 please review this, as i remember that you implemented #bytes

File: edsl.md [hunk 1]
```diff
@@ -153,9 +159,21 @@ where `F1 : F2 : F3 : F4` is the (two's complement) byte-array representation of
...
+ // --------------------------------------------------------------------
+    rule  #padRightToWidth(N, WS) => #padRightToWidthAux(N -Int #sizeWordStack(WS), WS, .WordStack)
+
+    syntax WordStack ::= #padRightToWidthAux ( Int , WordStack , WordStack ) [function]
+ // -----------------------------------------------------------------------------------
+    rule #padRightToWidthAux(0, WS, ZEROS) => WS ++ ZEROS
+
+    rule #padRightToWidthAux(N, WS, ZEROS) => #padRightToWidthAux(N -Int 1, WS, 0 : ZEROS)
+      requires N >Int 0
+
```
Comment (ehildenb) [inline edsl.md]: I think that `#padRightToWidth` should probably be put in `data.md` next to `#padToWidth` definition, @tomtomjhj.

**Comment by yzhang90** [comment]
I took a look at the ABI spec again and I think jerry’s impl is correct. Btw, maybe add [concrete] to the padRightToWidth rule?

**Comment by tomtomjhj** [comment]
@yzhang90 the contents inside `#byte` might be symbolic. [example](https://github.com/runtimeverification/verified-smart-contracts/blob/gnosis/gnosis/gnosis-spec.ini#L174)

**Comment by daejunpark** [comment]
@yzhang90 why `[concrete]`? the call data could be symbolic, right?

**Pull request merged**